### PR TITLE
DOCS-8763 Pipeline Visibility Setup Documentation Updates

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do? What is the motivation?

Refactors the Setup CI Provider instructions with consistent edits across all pages, such as adding an Advanced Configuration section and linking to the Search and Manage CI Pipelines page.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/continuous_integration/pipelines/azure.md [hunk 1]
```diff
@@ -37,50 +37,48 @@ Set up tracing on Azure Pipelines to gain real time insights into your CI/CD wor
...
 2. For each project, go to **Project settings > Service hooks** in Azure DevOps and select the green plus (+) icon to create a subscription.
 
-3. Create a new subscription to the `Datadog CI Visibility` service for each of the following webhook types:
+3. Create a subscription to the `Datadog CI Visibility` service for each of the following webhook types. These event types are required and must be enabled individually. 
+
     - **Run state changed**
     - **Run stage state changed**
     - **Run job state changed**
-    - **Run stage approval completed**
-    - **Run stage waiting for approval**
```
Comment (gturbat) [inline content/en/continuous_integration/pipelines/azure.md:51]: Why are these two types removed?

File: content/en/continuous_integration/pipelines/azure.md [hunk 2]
```diff
@@ -99,25 +100,31 @@ Example for enabling the hooks in specified projects:
     projectName1 projectName2
 ```
 
+The script supports environment variables `DD_API_KEY` and `DD_SITE`, and flags parameters `--dd-api-key` and `--dd-site`.
```
Comment (gturbat) [inline content/en/continuous_integration/pipelines/azure.md:103]: I think it was more useful before the scripts example, readers could miss is so down below

File: content/en/continuous_integration/pipelines/awscodepipeline.md [hunk 3]
```diff
@@ -100,20 +108,19 @@ To do this, add the `detail.pipeline` filter in the rule event pattern defined w
...
-The event pattern above sets up the integration only for the `first-pipeline` and `second-pipeline` pipelines.
+The event pattern sets up the integration only for the `first-pipeline` and `second-pipeline` pipelines.
 
 ### Correlate pipelines with tests
 
-If you are using [Test Visibility][8] and your pipeline contains one or more [AWS CodeBuild][9] actions to execute tests, you can correlate your tests
-with the related pipeline inside Datadog Pipeline Visibility. For instructions, refer to [Add the pipeline execution ID](#add-the-pipeline-execution-id-as-an-environment-variable).
+If you are using [Test Visibility][8] and your pipeline contains one or more [AWS CodeBuild][9] actions to execute tests, you can correlate your tests with the related pipeline inside Datadog Pipeline Visibility. For instructions, refer to [Add the pipeline execution ID](#add-the-pipeline-execution-id-as-an-environment-variable).
 
 ### Enable log correlation
```
Comment (gturbat) [inline content/en/continuous_integration/pipelines/awscodepipeline.md:117]:
```suggestion
### Collect job logs
```

File: content/en/continuous_integration/pipelines/circleci.md [hunk 4]
```diff
@@ -78,28 +84,30 @@ To bulk-configure hooks for your projects:
...
+
+If a `key:value` pair contains any commas, surround it with quotes. For example, to add `key1:value1,"key2: value with , comma",key3:value3`, the following string would need to be appended to the **Webhook URL**: `?tags=key1%3Avalue1%2C%22key2%3A+value+with+%2C+comma%22%2Ckey3%3Avalue3`.
+
+
 
 #### Integrate with Datadog Teams
+
 To display and filter the teams associated with your pipelines, add `team:<your-team>` as a custom tag. The custom tag name must match your [Datadog Teams][8] team handle exactly.
 
 ### Enable log collection
```
Comment (gturbat) [inline content/en/continuous_integration/pipelines/circleci.md:98]:
```suggestion
### Collect job logs
```

File: content/en/continuous_integration/pipelines/github.md [hunk 5]
```diff
@@ -50,67 +50,65 @@ The following GitHub versions are supported:
...
+### Disable GitHub Actions tracing
+
+To disable the CI Visibility GitHub Actions integration, make sure the GitHub app is no longer subscribed to the
+workflow job and workflow run events. To remove the events:
+
+1. Go to the [GitHub Apps][9] page.
+2. Click **Edit > Permission & events** on the relevant Datadog GitHub App (if you have multiple apps, you will have to repeat the process for each).
+3. Scroll to the **Subscribe to events** section, and make sure that **Workflow job** and **Workflow run** are not selected.
+
 ### Enable log collection
```
Comment (gturbat) [inline content/en/continuous_integration/pipelines/github.md:84]:
```suggestion
### Collect job logs
```

File: content/en/continuous_integration/pipelines/azure.md [hunk 6]
```diff
@@ -99,25 +100,31 @@ Example for enabling the hooks in specified projects:
...
+
+### Set custom tags 
+
+You can set custom tags for all pipeline and job spans from your Azure projects to improve traceability. For more information, see [Custom Tags and Measures][6].
+
 ### Enable job log collection
```
Comment (gturbat) [inline content/en/continuous_integration/pipelines/azure.md:109]:
```suggestion
### Collect job logs
```

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 7]
```diff
@@ -648,23 +656,15 @@ Re/Initialize Datadog-Plugin Agent Http Client
 TRACE -> http://<HOST>:<TRACE_PORT>/v0.3/traces
 {{< /code-block >}}
 
-## Install the Datadog Agent
-
-The Datadog Jenkins plugin can either report metrics through the Datadog Agent or directly to Datadog if an API key is provided. If you don't have a Datadog Agent running on the Jenkins controller instance, Datadog recommends installing it first by following the [Agent installation instructions][14]. Whether you choose to use Agentless mode or the Agent-based mode, you are **required** to use the plugin.
+## Configure the Agent for log collection
```
Comment (gturbat) [inline content/en/continuous_integration/pipelines/jenkins.md:659]: There is a duplicate here, see line 743

File: content/en/continuous_integration/pipelines/gitlab.md [hunk 8]
```diff
@@ -228,60 +208,53 @@ For failed GitLab pipeline executions, each error under the `Errors` tab within
...
+| `ci_quota_exceeded`              | provider     | CI quota exceeded.                                         |
+| `pipeline_loop_detected`         | user         | Pipeline loop detected.                                    |
+| `builds_disabled`                | user         | Build disabled.                                            |
+| `deployment_rejected`            | user         | Deployment rejected.                                      |
+| `protected_environment_failure`  | provider     | Environment failure.                                       |
+| `secrets_provider_not_found`     | user         | Secret provider not found.                                 |
+| `reached_max_descendant_pipelines_depth` | user   | Reached max descendant pipelines.                        |
+| `ip_restriction_failure`          | provider     | IP restriction failure.                                    |
 
 ## Enable job log collection
```
Comment (gturbat) [inline content/en/continuous_integration/pipelines/gitlab.md:246]:
```suggestion
### Collect job logs
```

File: content/en/continuous_integration/pipelines/azure.md (hunk 1 above)
Comment (alai97) [reply content/en/continuous_integration/pipelines/azure.md:51]: (Feel free to correct this if it's incorrect) but I was looking at the Azure setup in Org2 and Demo and didn't see these two:

Screenshot 2024-08-22 at 8 36 11 AM

File: content/en/continuous_integration/pipelines/azure.md (hunk 2 above)
Comment (alai97) [reply content/en/continuous_integration/pipelines/azure.md:103]:
```suggestion
```

File: content/en/continuous_integration/pipelines/azure.md [hunk 9]
```diff
@@ -37,50 +37,48 @@ Set up tracing on Azure Pipelines to gain real time insights into your CI/CD wor
...
 
 To run the script, you need:
 
 - An Azure DevOps username
-- An Azure DevOps [API Token][13]
+- An Azure DevOps [API token][13]
 - An Azure DevOps organization name
 
-The script only needs python3 and the requests package. For more information, run:
+For more information, you can run the following command:
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/azure.md:71]:
```suggestion

The script supports environment variables `DD_API_KEY` and `DD_SITE`, and flags parameters `--dd-api-key` and `--dd-site`.

For more information, you can run the following command:
```

File: content/en/continuous_integration/pipelines/jenkins.md (hunk 7 above)
Comment (alai97) [reply content/en/continuous_integration/pipelines/jenkins.md:659]: I'm just confirming that we're comfortable removing this [Configure the Agent for log collection](https://docs.datadoghq.com/continuous_integration/pipelines/jenkins/?tab=reportthroughthedatadogagentrecommended#configure-the-agent-for-log-collection) section with instructions for Linux, Windows, and macOS.

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 10]
```diff
@@ -738,7 +673,7 @@ If you are using Kubernetes to manage your Jenkins instances, add the `DD_CI_HOS
 
 This is only required for Jenkins workers. For the Jenkins controller, the infrastructure metric correlation does not require additional actions.
 
-**Note**: Infrastructure metric correlation is supported since Jenkins Plugin v5.0.0+
+**Note**: Infrastructure metric correlation is supported since Jenkins Plugin v5.0.0 or later.
 
 ## Enable job log collection
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/jenkins.md:678]:
```suggestion
## Collect job logs
```

File: content/en/continuous_integration/pipelines/azure.md (hunk 1 above)
Comment (gturbat) [reply content/en/continuous_integration/pipelines/azure.md:51]: It is actually the opposite, we miss them in-app. I will ask to add them

**Comment by gturbat** [comment]
There is a header indentation problem for Jenkins. Most entries in https://docs-staging.datadoghq.com/alai97/pipeline-visibility-setup-passthrough/continuous_integration/pipelines/jenkins/?tab=agentlessusinganapikey#install-the-datadog-jenkins-plugin-agentless should exist outside of https://docs.datadoghq.com/continuous_integration/pipelines/jenkins/?tab=reportthroughthedatadogagentrecommended#install-the-datadog-agent because they are available for agentless and agent modes

File: content/en/continuous_integration/pipelines/azure.md [hunk 11]
```diff
@@ -37,50 +37,51 @@ Set up tracing on Azure Pipelines to gain real time insights into your CI/CD wor
...
 1. Install the [Datadog CI Visibility][8] extension from the Azure Marketplace. There are several extensions starting with **Datadog**, make sure that you are installing the [Datadog CI Visibility][8] extension.
 
 2. For each project, go to **Project settings > Service hooks** in Azure DevOps and select the green plus (+) icon to create a subscription.
 
-3. Create a new subscription to the `Datadog CI Visibility` service for each of the following webhook types:
+3. Create a subscription to the `Datadog CI Visibility` service for each of the following webhook types. These event types are required and must be enabled individually. 
+
     - **Run state changed**
     - **Run stage state changed**
     - **Run job state changed**
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/azure.md:50]:
```suggestion
    - **Run job state changed**
    - **Run stage approval completed**
    - **Run stage waiting for approval**
```

File: content/en/continuous_integration/pipelines/circleci.md [hunk 12]
```diff
@@ -42,30 +42,36 @@ Set up tracing on CircleCI to optimize the performance of your pipelines, improv
 The Datadog integration for [CircleCI][1] works by using [webhooks][2] to send data to Datadog.
 
 1. For each project, go to **Project Settings > Webhooks** in CircleCI and add a new webhook:
-   * **Webhook URL**: <code>https://webhook-intake.{{< region-param key="dd_site" >}}/api/v2/webhook/?dd-api-key=<API_KEY></code> where `<API_KEY>` is your [Datadog API key][3].
-   * **Name**: `Datadog CI Visibility` or any other identifier name that you want to provide.
-   * **Events**: Select `Workflow Completed` and `Job Completed`.
+
+   * **Webhook Name**: `Datadog CI Visibility` or any other identifier name that you want to provide.
+   * **Receiver URL**: <code>https://webhook-intake.{{< region-param key="dd_site" >}}/api/v2/webhook/?dd-api-key=<API_KEY></code> where `<API_KEY>` is your [Datadog API key][3].
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/circleci.md:47]:
```suggestion
   * **Webhook URL**: <code>https://webhook-intake.{{< region-param key="dd_site" >}}/api/v2/webhook/?dd-api-key=<API_KEY></code> where `<API_KEY>` is your [Datadog API key][3].
```

File: content/en/continuous_integration/pipelines/github.md [hunk 13]
```diff
@@ -50,67 +50,65 @@ The following GitHub versions are supported:
...
+4. Alternatively, you can enable individual repositories by scrolling through the repository list and clicking the **Enable Job Logs Collection** toggle.
 
-If you are using self-hosted GitHub runners, you can correlate jobs to the host that is running them. To do this, make sure the GitHub runner name
-matches the hostname of the machine it is running on. CI Visibility uses this to link to
-infrastructure metrics. To see the metrics, click on a job span in the trace view and in the window a new tab named **Infrastructure** is
-shown which contains the host metrics.
+Immediately after toggling logs collection, workflow job logs are forwarded to Datadog Log Management. Log files larger than 1GiB are truncated.
 
-## Visualize pipeline data in Datadog
+**Note**: Logs are billed separately from CI Visibility. Log retention, exclusion, and indexes are configured in [Log Management][18]. Logs for GitHub jobs can be identified by the `datadog.product:cipipeline` and `source:github` tags.</div>
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/github.md:99]:
```suggestion
Logs are billed separately from CI Visibility. Log retention, exclusion, and indexes are configured in [Log Management][18]. Logs for GitHub jobs can be identified by the `datadog.product:cipipeline` and `source:github` tags.</div>
```

File: content/en/continuous_integration/pipelines/azure.md [hunk 14]
```diff
@@ -99,25 +105,30 @@ Example for enabling the hooks in specified projects:
...
 
-2. Add the Datadog app registration to your Azure DevOps organization and to every project for which you want to enable Log Collection. You can do this by going to "Organization settings" in your DevOps console. You may select "Add to all projects" to configure all projects in bulk.
+2. Add the Datadog app registration to your Azure DevOps organization and to each project you want to enable log collection. Navigate to **Organization settings** in your DevOps console, and select **Add to all projects** to configure all projects in bulk.
 
-<div class="alert alert-info"><strong>Note</strong>: Logs are billed separately from CI Visibility. Log retention, exclusion, and indexes are configured in Logs Settings. Logs for Azure jobs can be identified by the <code>datadog.product:cipipeline</code> and <code>source:azurepipelines</code> tags.</div>
+**Note**: Logs are billed separately from CI Visibility. Log retention, exclusion, and indexes are configured in [Log Management][18]. Logs for Azure jobs can be identified by the `datadog.product:cipipeline` and `source:azurepipelines` tags.
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/azure.md:125]:
```suggestion
Logs are billed separately from CI Visibility. Log retention, exclusion, and indexes are configured in [Log Management][18]. Logs for Azure jobs can be identified by the `datadog.product:cipipeline` and `source:azurepipelines` tags.
```

File: content/en/continuous_integration/pipelines/github.md [hunk 15]
```diff
@@ -50,67 +50,65 @@ The following GitHub versions are supported:
...
-To disable the CI Visibility GitHub Actions integration, make sure the GitHub app is no longer subscribed to the
-workflow job and workflow run events. To remove the events:
+## Visualize pipeline data in Datadog
 
-1. Go to the [GitHub Apps][9] page.
-2. Click **Edit > Permission & events** on the relevant Datadog GitHub App (if you have multiple apps, you will have to repeat the process for each).
-3. Scroll to the **Subscribe to events** section, and make sure that **Workflow job** and **Workflow run** are not selected.
+The [**CI Pipeline List**][7] and [**Executions**][8] pages populate with data after the pipelines finish.
 
+The **CI Pipeline List** page shows data for only the default branch of each repository. For more information, see [Search and Manage CI Pipelines][14].
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/github.md:111]:
```suggestion
The **CI Pipeline List** page shows data for only the default branch of each repository. For more information, see [Search and Manage CI Pipelines][19].
```

File: content/en/continuous_integration/pipelines/github.md [hunk 16]
```diff
@@ -133,3 +131,4 @@ workflow job and workflow run events. To remove the events:
 [15]: /glossary/#queue-time
 [16]: /glossary/#approval-wait-time
 [17]: /glossary/#custom-span
+[18]: /logs/guide/best-practices-for-log-management/
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/github.md:134]:
```suggestion
[18]: /logs/guide/best-practices-for-log-management/
[19]: /continuous_integration/search/#search-for-pipelines
```

**Review by maycmlee — COMMENTED** [review]
Just a couple of small edits! Also, for some reason this page isn't formatting correctly. I'm seeing the Language and Site dropdown menus all the way at the bottom left of the page: https://docs-staging.datadoghq.com/alai97/pipeline-visibility-setup-passthrough/continuous_integration/pipelines/github/

File: content/en/continuous_integration/pipelines/github.md [hunk 17]
```diff
@@ -50,67 +50,65 @@ The following GitHub versions are supported:
...
+4. Alternatively, you can enable individual repositories by scrolling through the repository list and clicking the **Enable Job Logs Collection** toggle.
 
-If you are using self-hosted GitHub runners, you can correlate jobs to the host that is running them. To do this, make sure the GitHub runner name
-matches the hostname of the machine it is running on. CI Visibility uses this to link to
-infrastructure metrics. To see the metrics, click on a job span in the trace view and in the window a new tab named **Infrastructure** is
-shown which contains the host metrics.
+Immediately after toggling logs collection, workflow job logs are forwarded to Datadog Log Management. Log files larger than 1 GiB are truncated.
 
-## Visualize pipeline data in Datadog
+Logs are billed separately from CI Visibility. Log retention, exclusion, and indexes are configured in [Log Management][18]. Logs for GitHub jobs can be identified by the `datadog.product:cipipeline` and `source:github` tags.</div>
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/github.md:99]:
```suggestion
Logs are billed separately from CI Visibility. Log retention, exclusion, and indexes are configured in [Log Management][18]. Logs for GitHub jobs can be identified by the `datadog.product:cipipeline` and `source:github` tags.
```

**Review by maycmlee — APPROVED** [review]

**Review by gturbat — APPROVED** [review]

**Pull request merged**